.evergreen/config.yml

```diff
@@ -1339,13 +1340,18 @@ buildvariants:
   - name: publish-packages
     distros:
     - ubuntu2004-small
-- name: publish-snapshot
+- name: publish
   display_name: "Publish"
   run_on: ubuntu1804-test
   tasks:
     - name: "upload-java"
-    - name: "publish-snapshot"
     - name: "upload-all"
+- name: java-release
+  display_name: Java Release
+  run_on: ubuntu1804-test
+  tasks:
+    - "publish-java-snapshots"
```

Comment:
Why do we need to publish the snapshot when we publish a release? I would think that the waterfall would have done that already.

Reply:
There are two tasks: 

- publish-java-snapshots
  which happens on non tagged pushes: `allow_for_git_tag: false`
  depends on upload_java - the latest java build
- publish-java-release
  Only happens on tag releases: `git_tag_only: true`
  Doesn't depend on upload_java - because it will use the canonical tagged assets

They both call the same function underneath so could be removed. But it does indicate in the evergreen build what task is run.

Its the gradle file that determines the s3 location for the assets to be packaged with the release - regardless of it being a snapshot or tagged release.

Reply:
As they both call the same underlying function - there probably isnt a benefit for the added complexity.  Updated to use a single task.